### PR TITLE
feat: Add MangoHud support to Steam images running Sway

### PR DESCRIPTION
**Pull request by shocklateboy92**
Since Mangoapp is already running in Gamescope, I figured [MangoHud](https://github.com/flightlessmango/MangoHud/) makes sense for Sway.

I have no idea *where* to document this. The current Wolf documenation doesn't mention MangoApp anywhere.

But in order for this to be useful, users need to know to press `Right Shift + F12` to make the overlay show up, and `Right Shift + F11` to change the position of the overlay (shows up in the top left by default).

### CODE REVIEW

**Comment by ABeltramo** [comment]
Thanks for the PR, I like the idea of adding `mangohud`! 

The good news is that `mangoapp` already brings in `mangohud` so the Steam image that we have **already bundles that in**, here's an example of running `mangohud vkgears`

Screenshot 2024-08-25 at 10 09 09

This works when I open up a terminal using docker, but when I try to run it from inside the session it tries to parse `/tmp/mangohud.uO7uX5um`
```cfg
control=mangohud
fsr_steam_sharpness=5
nis_steam_sharpness=10
no_display
```
if I comment out the `no_display` option I can successfully see the overlay

Screenshot 2024-08-25 at 10 22 43


There are still these 3 errors that we probably should look into
```
[2024-08-25 10:17:18.690] [MANGOHUD] [error] [cpu.cpp:461] Could not find cpu temp sensor location
[2024-08-25 10:17:18.732] [MANGOHUD] [error] [loader_nvctrl.cpp:39] Failed to open 64bit libXNVCtrl.so.0: libXNVCtrl.so.0: cannot open shared object file: No such file or directory
[2024-08-25 10:17:18.732] [MANGOHUD] [error] [nvctrl.cpp:45] XNVCtrl loader failed to load
```
but we can leave that for later.

## Where does that config file come from?

https://github.com/games-on-whales/gow/blob/b3ac0cc48331de1374a567183afc594a29610e2a/images/steam/scripts/startup.sh#L33-L37

This has been a long time ago, a time when we were using Gamescope instead of Sway and Steam wasn't even starting because of network issues. I think we should try removing `STEAM_USE_MANGOAPP` and that default mango config and see if everything still works.

## TODO

To recap, I think we should:

- [ ] Remove `/tmp/mangohud.uO7uX5um` and test it out
- [ ] Check if we can fix the `mangohud` reported errors
- [ ] Add this to the docs "somewhere"
- [ ] Test it in games!

On a side note: the `MANGOHUD=1` env variable only works for Vulkan apps, for Steam games we should generally ask users to add 
```
mangohud %command%
```
to the launch options.

**Comment by shocklateboy92** [comment]
> Thanks for the PR, I like the idea of adding `mangohud`!
> 
> The good news is that `mangoapp` already brings in `mangohud` so the Steam image that we have **already bundles that in**.

Damn, `mangoapp` was installed in the path, but not `mangohud`, which is what made me think they weren't dependencies. But I was [clearly wrong](https://packages.ubuntu.com/noble/mangoapp) and it works in a fresh image, so maybe I had just messed up my container from earlier experiments.

> This works when I open up a terminal using docker, but when I try to run it from inside the session it tries to parse `/tmp/mangohud.uO7uX5um`
> 
> ```haproxy-config
> control=mangohud
> fsr_steam_sharpness=5
> nis_steam_sharpness=10
> no_display
> ```
> 

So, that actually works out really well for us, because `no_display` causes it to be hidden on start but the _keyboard shortcut to show still works_! This means we can leave this on by default and just document somewhere that users can get a frame counter on all proton games at any time by just pressing `Right Shift + F12`! :smile: 

> There are still these 3 errors that we probably should look into
> 
> ```
> [2024-08-25 10:17:18.690] [MANGOHUD] [error] [cpu.cpp:461] Could not find cpu temp sensor location
> ```

That one worked on my machine, I think it's hardware specific

> ```
> [2024-08-25 10:17:18.732] [MANGOHUD] [error] [loader_nvctrl.cpp:39] Failed to open 64bit libXNVCtrl.so.0: libXNVCtrl.so.0: cannot open shared object file: No such file or directory
> [2024-08-25 10:17:18.732] [MANGOHUD] [error] [nvctrl.cpp:45] XNVCtrl loader failed to load
> ```

libXNVCtrl looks to be part of `nvidia-settings` package. We can install it if you really want GPU fan speed info. Though I'm okay without it so that the overlay doesn't get too massive by default.

I did get:
```
[2024-08-25 18:05:21.180] [MANGOHUD] [error] [overlay_params.cpp:741] Unknown option 'nis_steam_sharpness'
```

And I see `nis_steam_sharpness` in the generated mango config, but  I can't for the life of me figure out _where_ that comes from. I couldn't find a single reference to `nis_steam_sharpness` or even just `sharpness` in your entire repo. :confused: 

> 
> On a side note: the `MANGOHUD=1` env variable only works for Vulkan apps, for Steam games we should generally ask users to add
> 
> ```
> mangohud %command%
> ```
> 
> to the launch options.

Yep. I forgot to write that in the PR description, but that will also need to be included in the documentation. 
Speaking of which, where should we put this? Shall we move the `Steam` page from GOW to Wolf documentation?

Long term, we should merge those - most users won't think to look in both places.
